File: codechef/snackdown2016/MAKEART.js
const readline = require('readline');
const fs = require('fs');

process.stdin.resume();
process.stdin.setEncoding('utf8');

const rl = readline.createInterface({
    input: process.stdin,
    terminal: false
});

var lines = 0;
var T;

rl.on('line', function(data) {
    if (lines === 0) {
        // T = parseInt(data);
    } else {
        if (lines % 2 === 1) {
            // n = parseInt(data);
        } else {
            N = data.split(' ');
            executeTest(N);
        }
    }

    lines++;
});

function executeTest(N) {
    var seq = 0;
    for (i = 0; i < N.length - 2; i++) {
        if (N[i] === N[i + 1] && N[i] === N[i + 2]) {
            console.log('Yes');
            return 0;
        }
    }
    console.log('No');
}

File: codechef/snackdown2016/MAKEART.py
import sys

def exec_test (count, N):
    seq = 0
    for i in range(0, count - 2):
        if N[i] == N[i+1] and N[i] == N[i+2]:
            print "Yes"
            return
    
    print "No"
    return

lines = 0
T = 0
count = 0

for line in sys.stdin:
    if lines == 0:
        T = line.strip("\n")
    else:
        if lines % 2 == 1:
            count = int(line.strip("\n"))
        else:
            N = [int(x) for x in line.strip("\n").split(" ")]
            exec_test(count, N)            

    lines += 1
